crates/biome_js_semantic/src/semantic_model/reference.rs

```diff
@@ -58,9 +58,12 @@ impl Reference {
 
     /// Returns the scope of this reference
     pub fn scope(&self) -> Scope {
-        let id = self
-            .data
-            .scope(TextRange::new(self.range_start(), self.range_start()));
+        let start = self.range_start();
+        let id = self.data.scope(TextRange::new(
+            start,
+            // SAFETY: A reference name has at least a length of 1 byte.
+            start.checked_add(1.into()).unwrap(),
```

Comment:
Why use `checked_add()` if you're just going to `unwrap()` anyway?

```suggestion
            start + 1.into(),
```

Reply:
I got tricked by `TextSize`. `start + 1.into()` doesn't compile. However, `start + TextSize::from(1)` compiles.